sw/device/tests/alert_handler_lpg_clkoff_test.c

```diff
@@ -0,0 +1,743 @@
+  */
+  if (test_step_cnt == 1 * ARRAYSIZE(kPeripherals)) {
+    alert_handler_config_all_peripherals(/* ping_timeout*/ 256);
+  }
+
+  while (test_step_cnt < 2 * ARRAYSIZE(kPeripherals) - 1) {
+    // Each test phase consists of ARRAYSIZE(kPeripherals) steps
+    // At every step, a specific peripheral is tested
+    test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+    test_phase = test_step_cnt / ARRAYSIZE(kPeripherals);
+    peri_idx = test_step_cnt - (test_phase)*ARRAYSIZE(kPeripherals);
+    // LOG_INFO("Test phase = %d, Periphereal idx = %d, test_step_cnt = %d",
+    // test_phase, peri_idx, test_step_cnt);
+
+    busy_spin_micros(rnd_wait_times[peri_idx]);
+
+    // Disable the clock of the peripheral
+    if (kPeripherals[peri_idx].is_hintable) {
+      CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(
+          &clkmgr, kPeripherals[peri_idx].clk_index, kDifToggleDisabled));
+    } else {
+      CHECK_DIF_OK(dif_clkmgr_gateable_clock_set_enabled(
+          &clkmgr, kPeripherals[peri_idx].clk_index, kDifToggleDisabled));
+    }
+
+    // wait enough
+    if (kDeviceType == kDeviceFpgaCw310) {
+      // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+      // (2**6)*2*4*(2**16)*(400ns) = 12.8s
+      busy_spin_micros(1000 * 12800);
+    } else if (kDeviceType == kDeviceSimDV) {
+      // NUM_ALERTS*2*margin_of_safety*(2**DW)*(1/kClockFreqPeripheralHz)
+      // (2**6)*2*4*(2**3)*(40ns) = 160us
+      busy_spin_micros(160);
+    }
+
+    // Check the alert status
+    dif_alert_handler_local_alert_t check_local_alert =
+        kDifAlertHandlerLocalAlertAlertPingFail;
+    CHECK_DIF_OK(dif_alert_handler_local_alert_is_cause(
+        &alert_handler, check_local_alert, &is_cause));
+    CHECK(!is_cause, "Expected response: No alert_ping_fail! but we got %d",
+          is_cause);
+
+    if (kPeripherals[peri_idx].is_hintable) {
+      CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(
+          &clkmgr, kPeripherals[peri_idx].clk_index, kDifToggleEnabled));
+    } else {
+      CHECK_DIF_OK(dif_clkmgr_gateable_clock_set_enabled(
+          &clkmgr, kPeripherals[peri_idx].clk_index, kDifToggleEnabled));
+    }
+
+    // Increment the test counter
+    flash_ctrl_testutils_counter_increment(&flash_ctrl, kCounterTestSteps);
+
+    // Reset the device after last peripheral
+    // to unlock the alert handler for the next phase
+    if (test_step_cnt == (test_phase + 1) * ARRAYSIZE(kPeripherals) - 1) {
+      CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
+    }
+  }
+
+  /* TEST PHASE #2: TEST THE FATAL ALERTS
+
+    The timeout value is set to 256
+    None of the alerts should trigger ping_timeout_alert
+    TEST:
+    enable_alerts (all_peripherals, timeout=256)
+    for_each_peripheral
+      1- wait for random time
+      2- Enable the peripheral's clock
+      3- Trigger the fatal alert of the peripheral
+      4- Disable the clock of the peripheral
+      5- Wait for 1ms
```

Comment:
We should clear the cause bit on the alert handler side before re-enabling the clock of the peripheral, so that we can check that the alert sender continues to transmit after the peripheral comes back online.

Reply:
This is a good point, I was not aware of it, thanks :) 

After clearing the cause bit, the test starts to fail as you expect. Now, I need to use the `DV backdoor` [(previous comment)](https://github.com/lowRISC/opentitan/pull/14858#discussion_r993874235) to trigger the fatal alerts from the DV side.